example/advection/advection.cpp

```diff
@@ -0,0 +1,399 @@
+  Container<Real>& cout = pmb->real_containers.Get(stage_name[stage]);
+  Container<Real>& dudt = pmb->real_containers.Get("dUdt");
+  parthenon::Update::AverageContainers(cin, base, beta);
+  parthenon::Update::UpdateContainer(cin, dudt, beta*pmb->pmy_mesh->dt, cout);
+  return TaskStatus::success;
+}
+
+
+TaskList AdvectionDriver::MakeTaskList(MeshBlock *pmb, int stage) {
+  TaskList tl;
+  // we're going to populate our last with multiple kinds of tasks
+  // these lambdas just clean up the interface to adding tasks of the relevant kinds
+  auto AddMyTask = [&tl, pmb, stage, this] (BlockStageNamesIntegratorTaskFunc func, TaskID dep) {
+    return tl.AddTask<BlockStageNamesIntegratorTask>(func,dep,pmb,stage,stage_name,integrator);
+  };
+  auto AddContainerTask = [&tl] (ContainerTaskFunc func, TaskID dep, Container<Real>& rc) {
+    return tl.AddTask<ContainerTask>(func,dep,rc);
+  };
+  auto AddTwoContainerTask = [&tl] (TwoContainerTaskFunc func, TaskID dep, Container<Real>& rc1, Container<Real>& rc2) {
+    return tl.AddTask<TwoContainerTask>(func,dep,rc1,rc2);
+  };
+
+  TaskID none(0);
+  // first make other useful containers
+  if (stage == 1) {
+    Container<Real>& base = pmb->real_containers.Get();
+    pmb->real_containers.Add("dUdt", base);
+    for (int i=1; i<integrator->_nstages; i++) pmb->real_containers.Add(stage_name[i], base);
+  }
+
+  Container<Real>& sc0  = pmb->real_containers.Get(stage_name[stage-1]);
+  Container<Real>& sc1  = pmb->real_containers.Get(stage_name[stage]);
+  Container<Real>& dudt = pmb->real_containers.Get("dUdt");
+
+
+  auto start_recv = AddContainerTask([](Container<Real>& rc) {
+    rc.StartReceiving(parthenon::BoundaryCommSubset::all);
+    return TaskStatus::success;
+  }, none, sc1);
+
+  auto advect_flux = AddContainerTask(Advection::CalculateFluxes, none, sc0);
+
+  auto send_flux = AddContainerTask([](Container<Real>& rc) {
+    rc.SendFluxCorrection();
+    return TaskStatus::success;
+  }, advect_flux, sc0);
+
+  auto recv_flux = AddContainerTask([](Container<Real>& rc) {
+      if (!rc.ReceiveFluxCorrection()) return TaskStatus::fail;
+      return TaskStatus::success;
+    }, advect_flux, sc0);
+
+  // compute the divergence of fluxes of conserved variables
+  auto flux_div = AddTwoContainerTask([](Container<Real>& u, Container<Real>& du) {
+    parthenon::Update::FluxDivergence(u, du);
+    return TaskStatus::success;
+  }, recv_flux, sc0, dudt);
+
+  // apply du/dt to all independent fields in the container
+  auto update_container = AddMyTask(UpdateContainer, flux_div);
+
+  // update ghost cells
+  auto send = AddContainerTask([](Container<Real>& rc) {
+    rc.SendBoundaryBuffers();
+    return TaskStatus::success;
+  }, update_container, sc1);
+  auto recv = AddContainerTask([](Container<Real>& rc) {
+    if ( !rc.ReceiveBoundaryBuffers() ) return TaskStatus::fail;
+    return TaskStatus::success;
+  }, send, sc1);
+  auto setC = AddContainerTask([](Container<Real>& rc) {
```

Comment:
`setC` is a confusing name for this task

Reply:
renamed.